llvm/unittests/Frontend/OpenMPCompositionTest.cpp

```diff
@@ -37,3 +37,29 @@ TEST(Composition, GetCompoundConstruct) {
   Directive C6 = getCompoundConstruct({OMPD_parallel_for, OMPD_simd});
   ASSERT_EQ(C6, OMPD_parallel_for_simd);
 }
+
+TEST(Composition, IsLeafConstruct) {
+  ASSERT_TRUE(isLeafConstruct(OMPD_loop));
+  ASSERT_TRUE(isLeafConstruct(OMPD_teams));
+  ASSERT_FALSE(isLeafConstruct(OMPD_for_simd));
```

Comment:
Nit: Can you add one test case for a combined construct? e.g. `ASSERT_FALSE(isLeafConstruct(OMPD_parallel_for));`.